labgrid/resource/ethernetport.py

```diff
@@ -0,0 +1,195 @@
+
+        for (errorIndication, errorStatus, errorIndex, varBindTable) in hlapi.bulkCmd(
+                hlapi.SnmpEngine(),
+                hlapi.CommunityData('public'),
+                hlapi.UdpTransportTarget((self.hostname, 161)),
+                hlapi.ContextData(),
+                0, 20,
+                *[x[0] for x in vars],
+                lexicographicMode=False):
+            if errorIndication:
+                Exception("snmp error {}".format(errorIndication))
+            elif errorStatus:
+                Exception("snmp error {}".format(errorStatus))
+            else:
+                port = {}
+                for (key, val), (base, label) in zip(varBindTable, vars):
+                    index = key.getMibSymbol()[-1][0].prettyPrint()
+                    val = val.prettyPrint()
+                    if label == 'status':
+                        val = val.strip("'")
+                    port[label] = val
+                ports[port.pop('index')] = port
+
+        return ports
+
+    def _get_fdb(self):
+        from pysnmp import hlapi
+
+        ports = {}
+
+        for (errorIndication, errorStatus, errorIndex, varBindTable) in hlapi.bulkCmd(
+                hlapi.SnmpEngine(),
+                hlapi.CommunityData('public'),
+                hlapi.UdpTransportTarget((self.hostname, 161)),
+                hlapi.ContextData(),
+                0, 50,
+                hlapi.ObjectType(hlapi.ObjectIdentity('BRIDGE-MIB', 'dot1dTpFdbPort')),
+                lexicographicMode=False):
+            if errorIndication:
+                Exception("snmp error {}".format(errorIndication))
+            elif errorStatus:
+                Exception("snmp error {}".format(errorStatus))
+            else:
+                for varBinds in varBindTable:
+                    key, val = varBinds
+                    mac = key.getMibSymbol()[-1][0].prettyPrint()
+                    interface = str(int(val))
+                    ports.setdefault(interface, []).append(mac)
+
+        return ports
+
+    def update(self):
+        self.logger.debug("polling switch FDB")
+        self.fdb = self._get_fdb()
+        self.logger.debug("polling switch ports")
+        self.ports = self._get_ports()
+
+
+@attr.s
+class EthernetPortManager(ResourceManager):
+    def __attrs_post_init__(self):
+        super().__attrs_post_init__()
+        self.logger = logging.getLogger("{}".format(self))
+        self.loop = None
+        self.poll_task = None
+        self.switches = {}
+        self.neighbors = {}
+
+    def on_resource_added(self, resource):
+        if not isinstance(resource, SNMPEthernetPort):
+            return
+        self._start()
+        resource.avail=True
+
+    def _start(self):
+        import asyncio
+
+        if self.poll_task:
+            return
+
+        @asyncio.coroutine
+        def poll_step(self):
+            self.logger.debug("polling neighbor table")
+            self.neighbors = self._get_neigh()
```

Comment:
`self.neighbors` gets assigned here and is used only once later, maybe remove the instance variable and use it as a local variable in the later code flow.